src/cogent3/core/new_alignment.py

```diff
@@ -4333,7 +4380,6 @@ def no_degenerates(self, motif_length: int = 1, allow_gap: bool = False):
             character (latter is default, as most evolutionary modelling treats
             gaps as N).
         """
-        # refactor: array
         try:
             chars = list(self.moltype.alphabet)
         except AttributeError as e:
```

Comment:
I think every new moltype has an `.alphabet` attribute. Change this check to
```python
if self.moltype.degen_alphabet is None:
    raise MolTypeError(...)

chars = len(self.moltype)
```
Note I've also changed the exception type. We need a custom `MolTypeError` (which inherits from TypeError). That's what should be raised when, for example, someone tries to reverse complement a sequence that is not DNA or RNA.

Also, we need a test that triggers this condition.